Sources/MapboxMaps/Foundation/Camera/BasicCameraAnimator.swift

```diff
@@ -47,139 +62,170 @@ public class BasicCameraAnimator: NSObject, CameraAnimator, CameraAnimatorInterf
+        switch internalState {
+        case .initial:
+            fatalError("Attempt to continue an animation that has not started.")
+        case .inProgress:
+            propertyAnimator.continueAnimation(withTimingParameters: parameters, durationFactor: CGFloat(durationFactor))
+        case .final:
+            precondition(internalState != .final, "Attempt to continue an animation that has already completed.")
+        }
     }
 
-    internal var currentCameraOptions: CameraOptions? {
-
-        // Only call jumpTo if this animator is currently "active" and there are known changes to animate.
-        guard propertyAnimator.state == .active,
-              let transition = transition else {
-            return nil
+    func update() {
+        switch internalState {
+        case .initial, .final:
+            return
+        case let .inProgress(transition):
+            // The animator has been started or paused, so get the interpolated value. This may be nil if
+            // the animations haven't yet propagated into the CameraView's presentation tree.
+            if let presentationCameraOptions = cameraView.presentationCameraOptions {
+                mapboxMap.setCamera(
+                    to: cameraOptions(
+                        with: transition,
+                        cameraViewCameraOptions: presentationCameraOptions))
+            }
         }
+    }
 
+    private func cameraOptions(with transition: CameraTransition, cameraViewCameraOptions: CameraOptions) -> CameraOptions {
         var cameraOptions = CameraOptions()
-        let interpolatedCamera = cameraView.localCamera
 
         if transition.center.toValue != nil {
-            cameraOptions.center = interpolatedCamera.center?.wrap() // Wraps to [-180, +180]
+            cameraOptions.center = cameraViewCameraOptions.center?.wrap() // Wraps to [-180, +180]
         }
 
         if transition.bearing.toValue != nil {
-            cameraOptions.bearing = interpolatedCamera.bearing
+            cameraOptions.bearing = cameraViewCameraOptions.bearing
         }
 
         if transition.anchor.toValue != nil {
-            cameraOptions.anchor = interpolatedCamera.anchor
+            cameraOptions.anchor = cameraViewCameraOptions.anchor
         }
 
         if transition.padding.toValue != nil {
-            cameraOptions.padding = interpolatedCamera.padding
+            cameraOptions.padding = cameraViewCameraOptions.padding
         }
 
         if transition.zoom.toValue != nil {
-            cameraOptions.zoom = interpolatedCamera.zoom
+            cameraOptions.zoom = cameraViewCameraOptions.zoom
         }
 
         if transition.pitch.toValue != nil {
-            cameraOptions.pitch = interpolatedCamera.pitch
+            cameraOptions.pitch = cameraViewCameraOptions.pitch
         }
 
         return cameraOptions
     }
 
+    private func createTransition() {
```

Comment:
nit: do we want to use `makeTransition` instead of `createTransition` ?

Reply:
typically `make` prefix is for factory methods, but this method returns Void.

Reply:
ah i see, maybe we should name this something like `addTransitionToPropertyAnimator` ?

Reply:
or we could add a comment..